frontend/src/components/VersionExpirationWarning.vue

```diff
@@ -0,0 +1,154 @@
+            <span v-else>Version update will be enforced soon</span>
+          </template>
+          <span v-if="k8sExpiration.isError">Version update not possible due to missing update path. Please contact your landscape administrator</span>
+        </li>
+      </template>
+
+      <li v-for="({expirationDate, isValidTerminationDate, version, name, workerName, key, isInfo, isWarning, isError}) in expiredWorkerGroups" :key="key">
+        <span>Machine image <span class="font-weight-bold">{{name}} | Version: {{version}}</span> of worker group <span class="font-weight-bold">{{workerName}} </span></span>
+        <span v-if="isValidTerminationDate">expires <span class="font-weight-bold"><time-string :date-time="expirationDate"></time-string></span>. </span>
+        <span v-else>is expired. </span>
+        <span v-if="isInfo">Version will be updated in the next maintenance window</span>
+        <template v-if="isWarning">
+          <span v-if="isValidTerminationDate">Machine Image update will be enforced after that date</span>
+            <span v-else>Machine Image update will be enforced soon</span>
+        </template>
+        <span v-if="isError">Machine Image update as no newer version is available. Please choose another operating system</span>
+      </li>
+    </ul>
+  </g-popper>
+</template>
+
+<script>
+
+import TimeString from '@/components/TimeString'
+import GPopper from '@/components/GPopper'
+import find from 'lodash/find'
+import get from 'lodash/get'
+import { mapGetters } from 'vuex'
+import { shootItem } from '@/mixins/shootItem'
+import { k8sExpirationForShoot, expiredWorkerGroupsForShoot } from '@/utils'
+
+export default {
+  name: 'VerisonUpdateWarning',
+  components: {
+    TimeString,
+    GPopper
+  },
+  props: {
+    shootItem: {
+      type: Object
+    },
+    onlyK8sWarnings: {
+      type: Boolean
+    },
+    onlyMachineImageWarnings: {
+      type: Boolean
+    }
+  },
+  mixins: [shootItem],
+  computed: {
+    ...mapGetters([
+      'kubernetesVersions',
+      'machineImagesByCloudProfileName'
+    ]),
+    k8sExpiration () {
+      if (this.onlyMachineImageWarnings) {
+        return undefined
+      }
+      const k8sAutoPatch = get(this.shootItem, 'spec.maintenance.autoUpdate.kubernetesVersion', false)
+      return k8sExpirationForShoot(this.shootK8sVersion, this.shootCloudProfileName, k8sAutoPatch)
+    },
+    expiredWorkerGroups () {
+      if (this.onlyK8sWarnings) {
+        return []
+      }
+      const imageAutoPatch = get(this.shootItem, 'spec.maintenance.autoUpdate.machineImageVersion', false)
+      return expiredWorkerGroupsForShoot(this.shootWorkerGroups, this.shootCloudProfileName, imageAutoPatch)
+    },
+    isOverallStatusWarning () {
+      const isError = !!find([this.k8sExpiration, ...this.expiredWorkerGroups], { isError: true })
+      const isWarning = !!find([this.k8sExpiration, ...this.expiredWorkerGroups], { isWarning: true })
```

Comment:
use [some](https://lodash.com/docs/#some) instead of filter?

Reply:
yes - probably better operation to do this - done